### PR TITLE
Fixed bug in MergeInto function in storage/timestamp_cache

### PR DESCRIPTION
**Pull request by AbhishekSaha**
I saw this bug: https://github.com/cockroachdb/cockroach/issues/3554 and fixed the bug. I'm new to Cockroach and Open Source, so I'm still learning the ropes but I'm hoping this pull request is OK

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/3797)

### CODE REVIEW

**Comment by tschottdorf** [comment]
Thanks @AbhishekSaha! For the test, you could add something into `TestTimestampCacheMergeInto` in `storage/timestamp_cache_test.go` (or add a new test that exercises the logic you changed in a more focused way).

You can run only that test with `make test PKG=./storage TESTS=TestTimestampCacheMergeInto`.

**Comment by AbhishekSaha** [comment]
@tschottdorf:  I pushed the new test I wrote to my local branch, and the test succeeds, but that's not showing in this PR. Do I open a new PR for the test code?

**Comment by tschottdorf** [comment]
It seems that the commit is here, isn't it? Run `make check` locally to see why the tests failed - it's likely that you didn't run `gofmt` on some of your changes.

File: storage/timestamp_cache_test.go [hunk 1]
```diff
@@ -231,6 +231,14 @@ func TestTimestampCacheMergeInto(t *testing.T) {
 			if rTS, _ := tc2.GetMax(roachpb.Key("a"), roachpb.Key("c"), nil); !rTS.Equal(aaTS) {
 				t.Error("expected \"a\"-\"c\" to have aaTS timestamp")
 			}
+			//Proposed new test by AbhishekSaha
```
Comment (tamird) [inline storage/timestamp_cache_test.go]: i don't think these comments are useful, please remove them.

**Comment by tamird** [comment]
LGTM. Please squash this down to one commit when you're ready.

**Comment by tamird** [comment]
manually merged in 1c2b63b34dbe33a0d47a6d7a1a6987b5ad7d3fe2. Thanks!

**Pull request closed without merging**